packages/web-client/app/controllers/card-pay.ts

```diff
@@ -15,45 +16,53 @@ interface ChainChangeModalOptions {
   dismissable: boolean;
 }
 
-const networkCorrectionMessages = {
-  layer1: {
-    title: `Please connect to ${c.layer1.fullName}`,
-    body: `Card Pay uses ${c.layer1.fullName} as its Layer 1 network. To ensure the safety of your assets and transactions, this page will reload as soon as you change your wallet’s network to ${c.layer1.fullName} or disconnect your wallet.`,
-  },
-  layer2: {
-    title: `Please connect to ${c.layer2.fullName}`,
-    body: `Card Pay uses ${c.layer2.fullName} as its Layer 2 network. To ensure the safety of your assets and transactions, we’ve disconnected your wallet. You can restart any incomplete workflows after reconnecting with ${c.layer2.fullName}.`,
-  },
-};
-
 export default class CardPayController extends Controller {
   cardPayLogo = '/images/icons/card-pay-logo.svg';
   @service declare layer1Network: Layer1Network;
   @service declare layer2Network: Layer2Network;
   @tracked isShowingLayer1ConnectModal = false;
   @tracked isShowingLayer2ConnectModal = false;
   @tracked needsReload = false;
-  @tracked chainChangeModalOptions: ChainChangeModalOptions | null = null;
+  @tracked networkProblemModalOptions: NetworkProblemModalOptions | null = null;
 
   constructor() {
     super(...arguments);
 
     this.layer1Network.on('correct-chain', this.maybeReload);
     this.layer1Network.on('disconnect', this.maybeReload);
     this.layer1Network.on('incorrect-chain', this.onLayer1Incorrect);
+    this.layer1Network.on('websocket-disconnected', () =>
+      this.showWebsocketDisconnectedModal('layer1')
+    );
 
     this.layer2Network.on('incorrect-chain', this.onLayer2Incorrect);
+    this.layer2Network.on('websocket-disconnected', () =>
+      this.showWebsocketDisconnectedModal('layer2')
+    );
   }
 
   @action disconnectLayer1() {
     this.layer1Network.disconnect();
   }
 
+  @action showWebsocketDisconnectedModal(network: keyof typeof c) {
+    if (this.networkProblemModalOptions) return;
+    this.showNetworkProblemModal({
+      title: `Disconnected from ${c[network].fullName}`,
+      body: `Sorry! Card Pay is disconnected from ${c[network].fullName}. You can restore the connection by refreshing the page.`,
+      onClose: this.hideNetworkProblemModal,
+      action: () => window.open(config.urls.discordSupportChannelUrl, '_blank'),
+      actionText: 'Contact Cardstack Support',
+      dismissable: true,
+    });
+  }
+
   @action
   onLayer1Incorrect() {
     this.needsReload = true;
-    this.showChainChangeModal({
-      ...networkCorrectionMessages.layer1,
+    this.showNetworkProblemModal({
+      title: `Please connect to ${c.layer1.fullName}`,
+      body: `Card Pay uses ${c.layer1.fullName} as its Layer 1 network. To ensure the safety of your assets and transactions, this page will reload as soon as you change your wallet’s network to ${c.layer1.fullName} or disconnect your wallet.`,
```

Comment:
I have similar thoughts about this copy as well. I think it's better we lay out the problem more precisely, and be more direct with what we need the user to do. My suggestion:

"We detected your wallet is connected to X network, while Y network is needed to manage your Card Pay assets and transactions. Please change your wallet's network to Y, and then reload the page."

Reply:
I agree that friendlier/less terse messaging makes sense here!

Reply:
Added https://linear.app/cardstack/issue/CS-2468/improve-error-messages-shown-when-network-problems-occur